### ISSUE TITLE
KafkaBasedCommandSender might not receive command response if response topic didn't exist before

### ISSUE THREAD

**Issue by calohmn** [eclipse/hono#2741]
The `CommandAndControlAmqpIT#testSendCommandFailsWhenNoCredit` test sometimes fails:
````
[ERROR] testSendCommandFailsWhenNoCredit{AmqpCommandEndpointConfiguration, VertxTestContext}[2]  Time elapsed: 7.781 s  <<< FAILURE!
java.lang.AssertionError: 
java.lang.AssertionError: 
Expecting:
  "org.eclipse.hono.client.SendMessageTimeoutException: sending command timed out after 4600ms
	at org.eclipse.hono.tests.IntegrationTestSupport.lambda$sendCommand$21(IntegrationTestSupport.java:1133)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:942)
[...]
"
not to be an instance of: org.eclipse.hono.client.SendMessageTimeoutException
	at org.eclipse.hono.tests.amqp.CommandAndControlAmqpIT.lambda$testSendCommandFailsWhenNoCredit$69(CommandAndControlAmqpIT.java:763)
	at org.eclipse.hono.tests.amqp.CommandAndControlAmqpIT.lambda$testSendCommandFailsWhenNoCredit$70(CommandAndControlAmqpIT.java:758)
````

Log output:
````
2021-06-18T14:18:47.9336314Z 14:18:47.932 [main] INFO  o.e.h.t.amqp.CommandAndControlAmqpIT - running testSendCommandFailsWhenNoCredit(AmqpCommandEndpointConfiguration, VertxTestContext) [2]; parameters: subscribe as: GATEWAY_FOR_ALL_DEVICES, southbound endpoint: command, northbound endpoint: command
2021-06-18T14:18:47.9338917Z 14:18:47.932 [main] INFO  o.e.h.tests.IntegrationTestSupport - Configured to connect to Kafka on localhost:34277
2021-06-18T14:18:47.9345257Z 14:18:47.933 [main] INFO  o.e.h.tests.IntegrationTestSupport - Configured to connect to Kafka on localhost:34277
2021-06-18T14:18:47.9401644Z 14:18:47.939 [main] INFO  o.e.h.a.SpringBasedHonoPasswordEncoder - using BCrypt [strength: 4] with PRNG [NativePRNGNonBlocking] for encoding clear text passwords
2021-06-18T14:18:47.9482431Z 14:18:47.947 [main] INFO  o.e.h.a.SpringBasedHonoPasswordEncoder - using BCrypt [strength: 4] with PRNG [NativePRNGNonBlocking] for encoding clear text passwords
2021-06-18T14:18:47.9744999Z 14:18:47.973 [vert.x-kafka-consumer-thread-167] WARN  o.apache.kafka.clients.NetworkClient - [Consumer clientId=event-0ba547d8-fc4a-4948-b1f3-a84a83665433, groupId=its-9709fdaf-ef8c-4a1b-8425-12ffcb42ee14] Error while fetching metadata with correlation id 2 : {hono.event.a1bf5a70-2fc9-4c5a-a07f-5e470cd7985b=LEADER_NOT_AVAILABLE}
2021-06-18T14:18:48.1404370Z 14:18:48.139 [vert.x-eventloop-thread-2] INFO  o.e.h.t.amqp.CommandAndControlAmqpIT - AMQPS connection to adapter [localhost:49176] established
2021-06-18T14:18:51.0585766Z 14:18:51.057 [vert.x-eventloop-thread-1] WARN  o.e.h.a.c.k.i.KafkaBasedCommandSender - [auto.offset.reset] value is set to other than [latest]. It will be ignored and internally set to [latest]

2021-06-18T14:18:51.0783260Z 14:18:51.074 [vert.x-kafka-consumer-thread-168] WARN  o.apache.kafka.clients.NetworkClient - [Consumer clientId=command_response-ee0779bf-4c18-4f68-b53e-768c3f69ae60, groupId=a1bf5a70-2fc9-4c5a-a07f-5e470cd7985b-750e0d48-7f1b-4279-964d-c4fb9cad092a] Error while fetching metadata with correlation id 2 : {hono.command_response.a1bf5a70-2fc9-4c5a-a07f-5e470cd7985b=LEADER_NOT_AVAILABLE}

2021-06-18T14:18:51.8736854Z MQTT14:18:51.872 [kafka-producer-network-thread | Hono MQTT Adapter-event-172e7629-c175-4a8a-8e74-19096c42c3df] WARN  o.apache.kafka.clients.NetworkClient - [Producer clientId=Hono MQTT Adapter-event-172e7629-c175-4a8a-8e74-19096c42c3df] Error while fetching metadata with correlation id 1308 : {hono.event.ff745dc0-f404-4b97-a28a-7068366ec841=LEADER_NOT_AVAILABLE, hono.event.f6e0747c-0536-44f5-8767-787da6102d16=LEADER_NOT_AVAILABLE}

2021-06-18T14:18:55.6700723Z 14:18:55.666 [main] INFO  o.e.h.tests.IntegrationTestSupport - Configured to connect to Kafka on localhost:34277
2021-06-18T14:18:55.6817039Z 14:18:55.680 [vert.x-eventloop-thread-2] INFO  o.e.h.t.amqp.CommandAndControlAmqpIT - connection to AMQP adapter closed
````

**Comment by calohmn** [comment]
I could reproduce the issue locally. On sending a request/response command, the `KafkaBasedCommandSender` first creates a command response consumer. As the tenant-related Kafka topics don't get explicitly created in the integration tests, this means the topic will be auto-created when this topic subscription is created.
The
```
Error while fetching metadata with correlation id 2 : {hono.event.a1bf5a70-2fc9-4c5a-a07f-5e470cd7985b=LEADER_NOT_AVAILABLE}
```
warning is related to that.
As it seems, in these cases sometimes the partitions given in the initial partition assignment will then be empty.
Currently, `KafkaBasedCommandSender` just waits on the initial partition assignment (with no check regarding the assigned partitions) and then publishes the command message.
In the integration tests where this issue occurs, the command response gets published rather quickly.
In detailed logs of a failed test like above, one can see that only after the command response got published, another rebalance for the command response consumer occurs, this time with a non-empty partition list.
As the `auto.offset.reset` config of the consumer is set to `latest`, the consumer will only receive commands that are not yet published, meaning the already published command response message is skipped.

To fix this, the `KafkaBasedCommandSender` should wait for the 2nd rebalance with the non-empty partition list.

**Comment by calohmn** [comment]
**Another issue:**
There is another potential error case, where the command response consumer skips a command response record.
It occurs if the record is published *before* the command response consumer has fetched the committed offsets.
As the `auto.offset.reset` config of the consumer is set to `latest`, the consumer will then only handle records published after the said record.

A fix here would involve delaying the completion of the command response consumer `start()` future until after the committed offsets have been fetched.
Unfortunately, there is no hook for that exact point in time, so that it looks like we have to wait for the first records fetch operation to complete here, which will add quite some delay.
